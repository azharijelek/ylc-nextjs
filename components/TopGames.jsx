import Link from 'next/link'
import Button from '@material-ui/core/Button'

export default function TopGames() {
  const games = [
    {
      name: 'Trivia',
      permalink: '/fun/games/trivia',
      img:
        'https://cdn.statically.io/img/dev.yourlifechoices.com.au/f=auto%2Cq=100/wp-content/themes/bimber-child-theme/assets/img/games/trivia.png'
    },
    {
      name: 'Word Search',
      permalink: '/fun/games/word-search',
      img:
        'https://cdn.statically.io/img/dev.yourlifechoices.com.au/f=auto%2Cq=100/wp-content/themes/bimber-child-theme/assets/img/games/logo-wordsearch.png'
    },
    {
      name: 'cross-word',
      permalink: '/fun/games/aussie-crosswords',
      img:
        'https://cdn.statically.io/img/dev.yourlifechoices.com.au/f=auto%2Cq=100/wp-content/themes/bimber-child-theme/assets/img/games/crossword-logo.png'
    },
    {
      name: 'solitair',
      permalink: '/fun/games/solitaire',
      img:
        'https://cdn.statically.io/img/dev.yourlifechoices.com.au/f=auto%2Cq=100/wp-content/themes/bimber-child-theme/assets/img/games/solitair.png'
    }
  ]

  return (
    <>
      {games.map((game) => (
        <div className="game-item" key={game.name}>
          <Link href={game.permalink}>
            <a>
              <img src={game.img} alt={game.name} loading="lazy" />
            </a>
          </Link>
        </div>
      ))}

      <Link href="/fun/games">
        <Button
          component="a"
          href="/fun/games"
          fullWidth
          variant="outlined"
          color="primary"
          className="viewmoregames">
          View More Games
        </Button>
      </Link>

      <style jsx>{`
        .game-item {
          border-radius: 4px;
          background: #4f4f4f;
          height: 140px;
          text-align: center;
          padding: 0 35px;
          border-radius: 10px;
          position: relative;
          z-index: 0;
          margin-bottom: 10px;

          img {
            position: absolute;
            top: 50%;
            transform: translateY(-40%);
            left: 0;
            right: 0;
            margin: 0 auto;
          }
        }
      `}</style>
    </>
  )
}
