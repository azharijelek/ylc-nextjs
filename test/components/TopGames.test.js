import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, mount } from 'enzyme'
import preloadAll from 'jest-next-dynamic'

configure({ adapter: new Adapter() })

import TopGames from '@/components/TopGames'

beforeAll(async () => {
  await preloadAll()
})

//const logopath = '/static/logo.svg'
const wrapper = mount(<TopGames />)

describe('TopGames', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders 4 Images', () => {
    expect(wrapper.find('img')).toHaveLength(4)
  })

  it('renders 4 div containers', () => {
    expect(wrapper.find('div')).toHaveLength(4)
  })
})
