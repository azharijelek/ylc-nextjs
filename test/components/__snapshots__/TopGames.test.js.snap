// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopGames renders correctly 1`] = `ReactWrapper {}`;
